src/pexprs-madlibs.js

```diff
@@ -0,0 +1,209 @@
+'use strict';
+
+// --------------------------------------------------------------------
+// Imports
+// --------------------------------------------------------------------
+
+var common = require('./common');
+var pexprs = require('./pexprs');
+
+// --------------------------------------------------------------------
+// Operations
+// --------------------------------------------------------------------
+
+pexprs.PExpr.prototype.generateExample = common.abstract;
+
+pexprs.any.generateExample = function(grammar, examples, inSyntacticContext, actuals) {
+  return {example: String.fromCharCode(Math.floor(Math.random() * 255))};
+};
+
+function categorizeExamples(examples) {
+  var examplesNeeded = examples.filter(function(example) {
+                                  return example.hasOwnProperty('examplesNeeded');
+                                })
+                               .map(function(example) { return example.examplesNeeded; });
+  examplesNeeded = [].concat.apply([], examplesNeeded)
```

Comment:
I don't get what's going on here. It looks like `examplesNeeded` is always guaranteed to be an Array, so I'm not sure why you need to call `concat`, and anyways `[].concat.apply([], examplesNeeded)` does nothing but copy the contents of `examplesNeeded` into a new array. Am I missing something?